sandbox/auth/session/token.go

```diff
@@ -2,16 +2,40 @@ package session
 
 import (
 	"github.com/go-jose/go-jose/v3"
+	"github.com/go-jose/go-jose/v3/jwt"
 	"golang.org/x/oauth2"
 )
 
+// Tokens can generally be ID or access. It's a bit weird to have a struct
+// named "Token" that only contains an ID token. An alternative is to just rename
+// this struct to IDToken.
 type Token struct {
-	oauth2.Token // Embed and oauth2 token
-
+	// Are these fields needed?
 	Issuer string            `json:"issuer,omitempty"`
 	Keys   []jose.JSONWebKey `json:"keys,omitempty"`
 
 	IDToken string `json:"id_token,omitempty"`
 
 	idClaims *IDClaims
 }
+
+func TokenFromString(t string) (*Token, error) {
+	_, err := jwt.ParseSigned(t)
+	if err != nil {
+		return nil, err
+	}
+	tok := &Token{
+		IDToken: t,
+	}
+	claims := tok.IDClaims()
+	if claims != nil {
+		tok.Issuer = claims.Issuer
+	}
+	return tok, nil
+}
+
+func (t *Token) ToOauth2Token() *oauth2.Token {
+	return &oauth2.Token{
```

Comment:
If we end up agreeing on leaving the `oauth2.Token` field alone, then we can remove this function.

If we do keep it, I have a worry around losing information. The original `oauth2.Token` has the following fields that we'd be losing: `TokenType`, `RefreshToken`, `Expiry`

In particular, I believe that `RefreshToken` and `Expiry` is what make it possible for the underlying oauth library to automatically do token refreshes. I believe that with this code we don't get any refreshing anymore.

Reply:
Good point, I messed this up. Agree we should not lose that info.